renormalizer/mps/mp.py

```diff
@@ -591,16 +589,15 @@ def variational_compress(self, mpo=None, guess=None):
             
             mps._switch_direction()
             
-            # check if convergence
-            if isweep > 0 and percent == 0 and \
-                    mps.distance(mps_old) / np.sqrt(mps.dot(mps.conj()).real) < mps.compress_config.vrtol:
-                converged = True
-                break
+            # check convergence
+            if isweep > 0 and percent == 0:
+                error = mps.distance(mps_old) / np.sqrt(mps.dot(mps.conj()).real)
+                logger.info(f"Variation compress relative error: {error}")
+                if error < mps.compress_config.vrtol:
+                    logger.info("Variational compress is converged!")
+                    break
             
             mps_old = mps.copy()
-
-        if converged:
-            logger.info("Variational compress is converged!")
         else:
```

Comment:
I can not find the `if` corresponding to the `else` here. Please check it.

Reply:
This is a syntax for the `if` clause. The code within the clause is executed if the loop is not stopped by ``break``. Example:
```
# Determine if 7 is a prime number
n = 7
for x in range(2, n):
    if n % x == 0:
        print(n, 'equals', x, '*', n//x)
        break
else:
    # loop fell through without finding a factor
    print(n, ' is a prime number')
```
[Official document](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops)